scheduler/plugin/plugin_test.go

```diff
@@ -1807,3 +1889,56 @@ func requireError(t *testing.T, err error, expectedError string) {
 		t.Fatalf("unexpected error: want %v, have %v", want, have)
 	}
 }
+
+type waitingPod struct {
+	pod            *v1.Pod
+	pendingPlugins map[string]*time.Timer
+	mu             sync.RWMutex
+}
+
+func (wp *waitingPod) GetPod() *v1.Pod {
+	return wp.pod
+}
+
+func (wp *waitingPod) GetPendingPlugins() []string {
+	wp.mu.RLock()
+	defer wp.mu.RUnlock()
+	var plugins []string
+	for plugin := range wp.pendingPlugins {
+		plugins = append(plugins, plugin)
+	}
+	return plugins
+}
+
+func (wp *waitingPod) Allow(pluginName string) {
+	wp.mu.Lock()
+	defer wp.mu.Unlock()
+	if timer, ok := wp.pendingPlugins[pluginName]; ok {
+		timer.Stop()
+		delete(wp.pendingPlugins, pluginName)
+	}
+}
+
+func (wp *waitingPod) Reject(pluginName, msg string) {
+	wp.mu.Lock()
+	defer wp.mu.Unlock()
+	if timer, ok := wp.pendingPlugins[pluginName]; ok {
+		timer.Stop()
+		delete(wp.pendingPlugins, pluginName)
+	}
+}
+
+func makeTestWaitingPod(pod *v1.Pod, plugins map[string]*time.Timer) framework.WaitingPod {
+	return &waitingPod{
+		pod:            pod,
+		pendingPlugins: plugins,
+	}
+}
+
+// comparePods compares the UIDs of two v1.Pod objects
+func comparePods(pod1, pod2 *v1.Pod) bool {
```

Comment:
why do we need to create this function? can't we somehow just use the common comparing functions such as reflect.DeepEqual etc?